crossConnIntegrationTests/src/main/native/include/LifetimeWrappers.h

```diff
@@ -10,184 +10,76 @@
 
 namespace hlt {
 
-struct InterruptHandle {
+struct InterruptHandle : hal::Handle<HAL_InterruptHandle, HAL_CleanInterrupts> {
  public:
-  explicit InterruptHandle(int32_t* status) {
-    handle = HAL_InitializeInterrupts(status);
-  }
-  InterruptHandle(const InterruptHandle&) = delete;
-  InterruptHandle operator=(const InterruptHandle&) = delete;
-
-  InterruptHandle(InterruptHandle&&) = default;
-  InterruptHandle& operator=(InterruptHandle&&) = default;
-
-  ~InterruptHandle() { HAL_CleanInterrupts(handle); }
-
-  operator HAL_InterruptHandle() const { return handle; }
-
- private:
-  HAL_InterruptHandle handle = 0;
+  explicit InterruptHandle(int32_t* status)
+      : hal::Handle<HAL_InterruptHandle, HAL_CleanInterrupts>{
```

Comment:
```suggestion
      : Handle{
```
C++ [injects the unqualified class name of a template instantiation into the class scope of the instantiation and its subclasses](https://en.cppreference.com/w/cpp/language/injected-class-name), so we can shorten this (and the other constructors of Handle subclasses). Whether or not we should is up to you as a matter of style, but I thought I should let you know about this option.

Reply:
yep, forgot that was a thing